AutoGraph/WebSocketClient.swift

```diff
@@ -0,0 +1,126 @@
+import Foundation
+import Starscream
+
+public typealias WebSocketCompletionBlock = (Result<WebSocketClient.Events, Error>) -> Void
+
+open class WebSocketClient {
+    public typealias SerializedObject = Decodable
+
+    public enum Events {
```

Comment:
@rex-remind101 Okay I updated the reason for using our own `Events` because instead of passing back data we are passing back serialized object. 

Also this way we can record the headers and log when the socket gets connected and disconnected